misc/grimoirelab2sh

```diff
@@ -0,0 +1,165 @@
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Authors:
+#     Luis Cañas-Díaz <lcanas@bitergia.com>
+#     Miguel Ángel Fernández Sánchez <mafesan@bitergia.com>
+#
+
+from __future__ import unicode_literals
+
+import argparse
+import datetime
+import json
+import sys
+
+from sortinghat.exceptions import InvalidFormatError
+from sortinghat.parsing.grimoirelab import GrimoireLabParser
+
+
+GRIMOIRELAB2SH_DESC_MSG = \
+"""Export identities information from GrimoireLab files to Sorting Hat JSON format."""
+
+
+def main():
+    """Export identities information from GrimoireLab files"""
+
+    args = parse_args()
+
+    try:
+        parser = parse_grimoirelab_file(args.identities, args.organizations,
+                                        args.source)
+    except (IOError, UnicodeDecodeError, InvalidFormatError) as e:
+        raise RuntimeError(str(e))
+
+    j = to_json(parser.identities, parser.organizations, args.source)
+
+    try:
+        args.outfile.write(j)
+        args.outfile.write('\n')
+    except IOError as e:
+        raise RuntimeError(str(e))
+
+
+def parse_args():
+    """Parse arguments from the command line"""
+
+    parser = argparse.ArgumentParser(description=GRIMOIRELAB2SH_DESC_MSG)
+
+    parser.add_argument('-i', '--identities', type=argparse.FileType('r'),
+                        help='GrimoireLab profiles/identities mapping file')
+    parser.add_argument('-d', '--organizations', dest='organizations',
+                        type=argparse.FileType('r'),
+                        help='GrimoireLab domain to employer mapping file')
+    parser.add_argument('-s', '--source', dest='source', required=True,
+                        help='name of the source')
+    parser.add_argument('-o', '--outfile', nargs='?', type=argparse.FileType('w'),
+                        default=sys.stdout,
+                        help='Sorting Hat JSON output filename')
+
+    args = parser.parse_args()
+
+    if not (args.identities or args.organizations):
+        parser.error('No input file passed, add --domain-employer or --identities')
+
+    return args
+
+
+def parse_grimoirelab_file(identities, organizations, source):
+    """Parse GrimoireLab JSON file"""
+
+    content_id = read_file(identities) if identities else None
+
+    content_org = read_file(organizations) if organizations else None
+
+    try:
+        parser = GrimoireLabParser(content_id, content_org, source=source)
+    except ValueError:
+        s = "Error: Empty input file(s)\n"
+        sys.stdout.write(s)
+        sys.exit(0)
+
+    return parser
+
+
+def to_json(uidentities, organizations, source):
+    """Convert unique identities and organizations to Sorting Hat JSON format"""
+
+    uids = {}
+    orgs = {}
+
+    # Convert to dict objects
+    for uidentity in uidentities:
+        uuid = uidentity.uuid
+
+        uid = uidentity.to_dict()
+        uid['identities'].sort(key=lambda x: x['email'])
```

Comment:
Remove this call because `email` is not a required value. When I run the script, I obtain the next error if some emails values are not given:
```
Traceback (most recent call last):
  File "grimoirelab2sh", line 157, in <module>
    main()
  File "grimoirelab2sh", line 50, in main
    j = to_json(parser.identities, parser.organizations, args.source)
  File "grimoirelab2sh", line 111, in to_json
    uid['identities'].sort(key=lambda x: x['email'])
TypeError: unorderable types: str() < NoneType()
```